pkg/bgpv1/test/adverts_test.go

```diff
@@ -0,0 +1,360 @@
+			podCIDRs:    []string{},
+			expectedRouteEvents: []routeEvent{
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "10.1.0.0",
+					prefixLen:   16,
+					isWithdrawn: true,
+				},
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "aaaa::",
+					prefixLen:   64,
+					isWithdrawn: true,
+				},
+			},
+		},
+		{
+			description: "re-add pod CIDRs",
+			podCIDRs: []string{
+				"10.1.0.0/16",
+				"aaaa::/64",
+			},
+			expectedRouteEvents: []routeEvent{
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "10.1.0.0",
+					prefixLen:   16,
+					isWithdrawn: false,
+				},
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "aaaa::",
+					prefixLen:   64,
+					isWithdrawn: false,
+				},
+			},
+		},
+		{
+			description: "update pod CIDRs",
+			podCIDRs: []string{
+				"10.2.0.0/16",
+				"bbbb::/64",
+			},
+			expectedRouteEvents: []routeEvent{
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "10.1.0.0",
+					prefixLen:   16,
+					isWithdrawn: true,
+				},
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "10.2.0.0",
+					prefixLen:   16,
+					isWithdrawn: false,
+				},
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "aaaa::",
+					prefixLen:   64,
+					isWithdrawn: true,
+				},
+				{
+					sourceASN:   ciliumASN,
+					prefix:      "bbbb::",
+					prefixLen:   64,
+					isWithdrawn: false,
+				},
+			},
+		},
+	}
+
+	testCtx, testDone := context.WithTimeout(context.Background(), maxTestDuration)
+	defer testDone()
+
+	// setup topology
+	gobgpPeers, fixture, cleanup, err := setup(testCtx, []gobgpConfig{gobgpConf}, fixtureConf)
+	require.NoError(t, err)
+	require.Len(t, gobgpPeers, 1)
+	defer cleanup()
+
+	// setup neighbor
+	err = setupSingleNeighbor(testCtx, fixture)
+	require.NoError(t, err)
+
+	// wait for peering to come up
+	err = gobgpPeers[0].waitForSessionState(testCtx, []string{"ESTABLISHED"})
+	require.NoError(t, err)
+
+	tracker := fixture.fakeClientSet.KubernetesFakeClientset.Tracker()
+
+	for _, step := range steps {
+		t.Run(step.description, func(t *testing.T) {
+			// update node spec
+			fixture.config.node.Spec.PodCIDRs = step.podCIDRs
+
+			err := tracker.Update(corev1.SchemeGroupVersion.WithResource("nodes"), fixture.config.node.DeepCopy(), "")
+			require.NoError(t, err, step.description)
+
+			// validate expected result
+			receivedEvents, err := gobgpPeers[0].getRouteEvents(testCtx, len(step.expectedRouteEvents))
+			require.NoError(t, err, step.description)
+
+			// match events in any order
+			require.ElementsMatch(t, step.expectedRouteEvents, receivedEvents, step.description)
+		})
+	}
+}
+
+// Test_LBEgressAdvertisement validates Service v4 and v6 IPs is advertised, withdrawn and modified on changing policy.
+func Test_LBEgressAdvertisement(t *testing.T) {
+	testutils.PrivilegedTest(t)
+
+	var steps = []struct {
+		description         string
+		srvName             string
+		ingressIP           string
+		oper                string // add or update
```

Comment:
Nit: What's `oper` stands for? Maybe "Operation"? Then it's common to use `op` I think.